app/models/miq_queue.rb

```diff
@@ -123,6 +123,66 @@ def self.put(options)
     msg
   end
 
+  # Trigger a background job
+  #
+  # target_worker:
+  #
+  # @options options [String] :class
+  # @options options [String] :instance
+  # @options options [String] :method
+  # @options options [String] :args
+  # @options options [String] :target_id (deprecated)
+  # @options options [String] :data (deprecated)
+  #
+  # execution parameters:
+  #
+  # @options options [String] :expires_on
+  # @options options [String] :ttl
+  # @options options [String] :task_id (deprecated)
+  #
+  # routing:
+  #
+  # @options options [String] :service name of the service. Similar to previous role or queue name derives
+  #                                    queue_name, role, and zone.
+  # @options options [ExtManagementSystem|Nil|Array<Class,id>] :resource resource for affinity. Typically an ems
```

Comment:
I'm ok with this for now, but I'm thinking resource might be a confusing name for the parameter...wondering if `:affinity` might be better because it much more clearly expresses the intent